docs/csharp/language-reference/operators/switch.md

```diff
@@ -0,0 +1,49 @@
+---
+title: "switch expressions - C# reference"
+description: Learn how to use the C# switch expression for pattern matching and other data introspection
+ms.date: 03/19/2020
+---
+# switch expressions (C# reference)
+
+This article covers the `switch` expression, introduced in C# 8.0. For information on the `switch` statement, see the article on the [`switch` statement](../statements/switch.md) in the [statements](../statements/index.md) section.
+
+## Basic example
+
+The `switch` expression provides for `switch`-like semantics in an expression context. It provides a concise syntax when the switch arms produce a value. The following example shows the structure of a switch expression. It translates values from an `enum` representing visual directions in an online map to the corresponding cardinal direction:
+
+:::code language="csharp" source="snippets/SwitchExpressions.cs" id="SnippetBasicStructure" interactive="csharp-interactive":::
+
+The preceding sample shows the basic elements of a switch expression:
+
+- The *range expression*: The preceding example uses the variable `direction` as the range expression.
```

Comment:
Is it a widely-used name: *range expression*? It has several meanings: I would call `1..^1` also a range expression. If we consider `switch` as operator, then it's *argument expression*.

Reply:
That's a good point. Right now, that's the term used in the [recursive patterns spec](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-8.0/patterns)  for the grammar.

I think this may change. The spec currently uses the term "switch-expression" for the same concept in the `switch` statement.

For now though, this is consistent.